src/MaskingContainer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Dhii\Container;
+
+use Dhii\Container\Exception\NotFoundException;
+use Dhii\Container\Util\StringTranslatingTrait;
+use Psr\Container\ContainerInterface;
+use function array_key_exists;
+
+/**
+ * An implementation of a container that wraps around another to selectively expose or mask certain keys.
+ *
+ * @since [*next-version*]
+ */
+class MaskingContainer implements ContainerInterface
+{
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $inner;
+
+    /**
+     * @var bool[]
+     */
+    protected $mask;
+
+    /**
+     * @var bool
+     */
+    protected $defMask;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $inner   The container whose entries to mask.
+     * @param bool[]             $mask    A mapping of keys to booleans, such that `true` exposes the mapped key and
+     *                                    `false` hides the mapped key.
+     * @param bool               $default The default mask value to use for keys that are not included in the mask.
+     */
+    public function __construct(ContainerInterface $inner, array $mask, $default = true)
```

Comment:
It was hard to understand how `$default` is used. Perhaps, a description of how it affects the result would help.

Reply:
Right! I'll add a better description.

For the record: it's a default mask, so you can set a default mask of `true` to hide everything then expose certain keys, or a default mask of `false` to expose everything and then hide certain keys.